enhancements/per-node-configuration.md

```diff
@@ -0,0 +1,850 @@
+
+Different security zones have different passthrough configurations:
+
+```yaml
+# L3Passthrough for DMZ nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3Passthrough
+metadata:
+  name: dmz-passthrough
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      security-zone: dmz
+  hostsession:
+    asn: 64512
+    hostasn: 64710
+---
+# L3Passthrough for internal nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3Passthrough
+metadata:
+  name: internal-passthrough
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      security-zone: internal
+  hostsession:
+    asn: 64512
+    hostasn: 64720
+```
+
+#### Multiple Instances on Same Node
+
+Example showing multiple L3VNIs configured on the same set of nodes for different tenants:
+
+```yaml
+# Tenant A VNI on compute nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3VNI
+metadata:
+  name: tenant-a-vni
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+  vrf: tenant-a
+  vni: 5001
+  hostsession:
+    asn: 64512
+    hostasn: 64600
+---
+# Tenant B VNI on the same compute nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3VNI
+metadata:
+  name: tenant-b-vni
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+  vrf: tenant-b
+  vni: 5002
+  hostsession:
+    asn: 64512
+    hostasn: 64601
+---
+# Tenant C VNI on the same compute nodes
+apiVersion: openpe.openperouter.github.io/v1alpha1
+kind: L3VNI
+metadata:
+  name: tenant-c-vni
+  namespace: openperouter-system
+spec:
+  nodeSelector:
+    matchLabels:
+      node-role.kubernetes.io/worker: ""
+  vrf: tenant-c
+  vni: 5003
+  hostsession:
+    asn: 64512
+    hostasn: 64602
+```
+
+In this example, all compute nodes (with `node-role.kubernetes.io/worker` label) will have three L3VNIs configured, one for each tenant. This is allowed and expected behavior for L3VNI, L2VNI, and L3Passthrough resources.
+
+### Controller Implementation
+
+#### Node Matching Logic
+
+All controllers (Underlay, L3VNI, L2VNI, and L3Passthrough) will be enhanced with:
+
+1. **Node Label Watching**: Add a watch for Node resource label changes to trigger reconciliation
+2. **Selector Matching**: For each node, determine which resource instance applies by matching node labels against all resource `nodeSelector` fields
+3. **Configuration Application**: Generate and apply configuration only on nodes matched by the resource's selector
+4. **Dynamic Updates**: Reconcile when:
+   - Resources are created/updated/deleted
+   - Node labels change
+   - Nodes are added/removed from the cluster
+
+#### Conflict Resolution
+
+**Underlay-specific restriction:**
+
+Only one Underlay can match a given node at any time. The validation webhook will prevent conflicts at admission time:
```

Comment:
the validation must happen also in the reconcile loop (and we have a pattern to do that). This is because webhooks might let configurations slip in in case of parallel creation of resources.